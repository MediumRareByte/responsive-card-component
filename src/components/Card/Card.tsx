import * as React from "react"
import styled from "styled-components"

interface ContainerProps {
  readonly maxWidth?: string
}
interface CardProps extends ContainerProps {
  readonly background: string
  readonly logo?: string
  readonly text?: string
}

const Container = styled.div<ContainerProps>`
  line-height: 0;
  max-width: ${(props): ContainerProps["maxWidth"] => props.maxWidth || "100%"};
  position: relative;
`
const CardImage = styled.img`
  width: 100%;
`
const CardOverlay = styled.div`
  background-color: rgba(0, 0, 0, 0.7);
  bottom: 0;
  display: flex;
  height: 2.9rem;
  position: absolute;
  width: 100%;
`
const LogoFlex = styled.div`
  box-sizing: border-box;
  display: flex;
  flex: 0 1 auto;
  padding: 0.3rem;
  position: relative;
  width: 2.9rem;
`
const Logo = styled.img`
  flex: 1 1 auto;
  width: 100%;
`
const TextFlex = styled.div`
  color: white;
  display: flex;
  flex: 1 1 auto;
  font-size: 1rem;
  font-weight: bold;
  line-height: 1.2rem;
`
const TextContainer = styled.span`
  margin: auto;
  flex: 1 1 auto;
`

export const Card: React.FC<CardProps> = (props: CardProps) => (
  <Container maxWidth={props.maxWidth}>
    <CardImage src={props.background}></CardImage>
    <CardOverlay>
      <LogoFlex>
        <Logo src={props.logo}></Logo>
      </LogoFlex>
      <TextFlex>
        <TextContainer>{props.text}</TextContainer>
      </TextFlex>
    </CardOverlay>
  </Container>
)
