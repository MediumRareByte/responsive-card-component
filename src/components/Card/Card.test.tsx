import React from "react"
import renderer from "react-test-renderer"

import { Card } from "./Card"

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <Card
        maxWidth="100%"
        background="bg.jpg"
        text="Home and Away"
        logo="logo.png"
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it("renders with text", () => {
  const instance = renderer.create(
    <Card
      maxWidth="100%"
      background="bg.jpg"
      text="Home and Away"
      logo="logo.png"
    />
  ).root
  const textContainer = instance.findAllByType("span")[0]

  expect(textContainer.children).toEqual(["Home and Away"])
})

it("renders without text", () => {
  const instance = renderer.create(
    <Card maxWidth="100%" background="bg.jpg" logo="logo.png" />
  ).root
  const textContainer = instance.findAllByType("span")[0]

  expect(textContainer.children).toEqual([])
})
