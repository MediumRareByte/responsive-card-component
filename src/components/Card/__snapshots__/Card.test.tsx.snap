// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-AxjAm byIbcN"
>
  <img
    className="sc-AxirZ cUVYWJ"
    src="bg.jpg"
  />
  <div
    className="sc-AxiKw jPfAJG"
  >
    <div
      className="sc-AxhCb gghXhC"
    >
      <img
        className="sc-AxhUy gNfBHr"
        src="logo.png"
      />
    </div>
    <div
      className="sc-AxgMl a-DHAA"
    >
      <span
        className="sc-AxheI cjAsIU"
      >
        Home and Away
      </span>
    </div>
  </div>
</div>
`;
